src/software/jetson_nano/services/motor.cpp

```diff
@@ -102,54 +111,223 @@ MotorService::MotorService(const RobotConstants_t& robot_constants,
     OPEN_SPI_FILE_DESCRIPTOR(front_right, FRONT_RIGHT_MOTOR_CHIP_SELECT)
     OPEN_SPI_FILE_DESCRIPTOR(back_left, BACK_LEFT_MOTOR_CHIP_SELECT)
     OPEN_SPI_FILE_DESCRIPTOR(back_right, BACK_RIGHT_MOTOR_CHIP_SELECT)
-    OPEN_SPI_FILE_DESCRIPTOR(dribbler, DRIBBLER_MOTOR_CHIP_SELECT)
+    // TODO (#2606) - Add Dribbler Motor support
 
     // Make this instance available to the static functions above
     g_motor_service = this;
+
+    // Clear faults
+    reset_gpio.setValue(GpioState::LOW);
+    sleep(1);
+    reset_gpio.setValue(GpioState::HIGH);
+    sleep(1);
+
+    // TMC6100 Setup
+    startDriver(FRONT_LEFT_MOTOR_CHIP_SELECT);
+    startDriver(BACK_RIGHT_MOTOR_CHIP_SELECT);
+    startDriver(FRONT_RIGHT_MOTOR_CHIP_SELECT);
+    startDriver(BACK_LEFT_MOTOR_CHIP_SELECT);
+
+    // Check faults
+    checkDriverFault(FRONT_LEFT_MOTOR_CHIP_SELECT);
+    checkDriverFault(FRONT_RIGHT_MOTOR_CHIP_SELECT);
+    checkDriverFault(BACK_LEFT_MOTOR_CHIP_SELECT);
+    checkDriverFault(BACK_RIGHT_MOTOR_CHIP_SELECT);
+
+    // TMC4671 Setup
+    startController(FRONT_LEFT_MOTOR_CHIP_SELECT);
+    startController(BACK_RIGHT_MOTOR_CHIP_SELECT);
+    startController(FRONT_RIGHT_MOTOR_CHIP_SELECT);
+    startController(BACK_LEFT_MOTOR_CHIP_SELECT);
 }
 
 MotorService::~MotorService() {}
 
-std::unique_ptr<TbotsProto::DriveUnitStatus> MotorService::poll(
-    const TbotsProto::DirectControlPrimitive& direct_control)
+
+bool MotorService::checkDriverFault(uint8_t motor)
 {
-    // TODO (#2335) We can only spin 1 motor right now
-    CHECK(encoder_calibrated_[FRONT_LEFT_MOTOR_CHIP_SELECT])
-        << "Running without encoder calibration can cause serious harm";
+    int gstat = tmc6100_readInt(motor, TMC6100_GSTAT);
+    std::bitset<32> gstat_bitset(gstat);
 
-    switch (direct_control.wheel_control_case())
+    CHECK(gstat_bitset.any() == false)
+        << "TMC6100 offline, no fault bits detected."
+        << " The reset bit should be set. This could indicate that the driver isn't"
+        << " communicating over SPI or the reset pin is not connected.";
+
+    if (gstat_bitset.any())
     {
-        case TbotsProto::DirectControlPrimitive::WheelControlCase::kDirectPerWheelControl:
-        {
-            // TODO (#2456) until we figure out the right factor considering
-            // the gear ratio
-            static float RANDOM_SCALING_FACTOR = 100.0;
+        LOG(WARNING) << "======= Faults For Motor " << std::to_string(motor)
+                     << "=========";
+    }
+
+    if (gstat_bitset[0])
+    {
+        LOG(WARNING)
+            << "Indicates that the IC has been reset. All registers have been cleared to reset values."
+            << "Attention: DRV_EN must be high to allow clearing reset";
+    }
+
+    if (gstat_bitset[1])
+    {
+        LOG(WARNING)
+            << "drv_otpw : Indicates, that the driver temperature has exceeded overtemperature prewarning-level."
+            << "No action is taken. This flag is latched.";
```

Comment:
seems like something that should be reported as a robot log back to thunderscope?

Reply:
it will be, all g3logs will make it back to tscope